### ISSUE TITLE
The type name 'Empty' does not exist in the type 'Guid'

### ISSUE THREAD

**Issue by wx2486** [dotnet/roslyn#71504]
**Version Used**: MSBuild version 17.7.4+3ebbd7c49 for .NET

**Steps to Reproduce**:

Compile the following piece of code:

```csharp
var s = string.Empty switch
{
    Guid.Empty.ToString() => null,
};
```

A minimal repro, with source-code provided, is ideal.  Using [sharplab](https://sharplab.io/) is preferred for compiler/language issues whenever possible.

**Diagnostic Id**:

If this is a report about a bug in an analyzer, please include the diagnostic if possible (e.g. `"IDE0030"`).

**Expected Behavior**:
Build succeeds.
**Actual Behavior**:
Got error "error CS0426: The type name 'Empty' does not exist in the type 'Guid'".

**Comment by wx2486** [comment]
The result doesn't change even with "using System;" added.

**Comment by wx2486** [comment]
It seems the expected behavior should be getting an error of "CS9135	A constant value of type 'string' is expected".

**Comment by jaredpar** [comment]
> It seems the expected behavior should be getting an error of "CS9135 A constant value of type 'string' is expected".

That is not the case though. In this situation the `Empty` can only be a valid type name due to the `.ToString` identifier that came after. Hence the compiler has issued the correct error here.
